src/clj/rems/api/services/attachment.clj

```diff
@@ -43,21 +44,23 @@
       (throw-forbidden))
     (attachments/save-attachment! file user-id application-id)))
 
-(defn zip-attachments [application]
-  (let [out (ByteArrayOutputStream.)]
+(defn zip-attachments [application all?]
+  (let [classes (model/classify-attachments application)
```

Comment:
I'm not that happy with calling model code here, but it was the simplest solution. I'd've preferred storing this information under `::attachment-classes` in the application. However, that presented a problem:

- zip-attachments needs the output of get-application-for-user because we want an application for which rems.application.model/hide-attachments has been done
- Currently, get-application-for-user is what gets otput from the API, so I can't add stuff there without it showing up in the API.
- I could add `::attachment-classes` to the output of get-application-internal (similarly to `::latest-decision-request-by-user`), but then I'd have to do the equivalent of hide-attachments here anyway

Reply:
I think overall we should have some concept of `raw` or `internal` vs. `public` or `external`. And be consistent about it. However we do have role-based authorization and data filtering so I think it can only be at the level of filtering done or not done. Also we have the db format which we mangle to get the internal format so I guess we have three. This could be something to talk about sometime that what could be the future big picture / if we could rewrite it.

I think simply reusing some code from somewhere is an acceptable solution here and some rework in the model the cleanest but more work and a future idea.